libasn1compiler/asn1c_misc.c

```diff
@@ -51,41 +51,55 @@ asn1c_make_identifier(enum ami_flags_e flags, asn1p_expr_t *expr, ...) {
 	char *str;
 	char *nextstr;
 	char *first = 0;
-	char *second = 0;
-	ssize_t size;
+	ssize_t size = 0;
 	char *p;
+	char *prefix = NULL;
+	char *sptr[4], **psptr = &sptr[0];
+	int sptr_cnt = 0;
 
 	if(expr) {
 		/*
 		 * Estimate the necessary storage size
 		 */
 		if(expr->Identifier == NULL)
 			return "Member";
-		size = strlen(expr->Identifier);
+		/*
+		 * Add MODULE name to resolve clash
+		 */
+		if(expr->_mark & TM_NAMECLASH) {
+			size += strlen(expr->module->ModuleName) + 2;
```

Comment:
When there are type names clash in different modules detected in `asn1f_check_duplicate()` of `asn1fix.c`, I add module name as prefix to the type name when making identifier for them : 

```
M1 DEFINITIONS ::= BEGIN
V ::= INTEGER
END

M2 DEFINITIONS ::= BEGIN
V ::= OCTET STRING
END
```
The constructed type name becomes `M1_V` and `M2_V` instead `V` and `V`.
So space for one underscore character '_' and one null terminate character `\0` equals 2.

By the way, I think `TM_NAMECLASH` is not a good name (due to my poor English and limited vocabulary) and I am not sure whether `expr->_mark` is a good place to hold the information what this `expr`'s name has duplication. 
Please feel free to improve them at your solely discretion.  Thanks.